### ISSUE TITLE
App doesn't stop in Visual Studio Code

### ISSUE THREAD

**Issue by satial-interfaces** [AvaloniaUI/Avalonia#9522]
When running an Avalonia 11 preview 4 desktop app in debug mode from Visual Studio Code, exiting it (the app) will not terminate it. It hangs in one of the threads (it seems). This does not happen if running the app from the command line in debug mode. The app needs to be force closed to be able to shut it down.

To Reproduce, do the following

1. Create the app-mvvm template-based app
2. Replace the 0.10.18 references with 11.0.0-preview4
3. Run it
4. Close it 
5. It keeps running

Expected behavior: it should terminate the app properly and give back result 0.

- OS: macOS Ventura (on Apple Silicon)
- Version 11.0.0-preview4
- Visual Studio Code 1.73.1

**Comment by maxkatz6** [comment]
> It hangs in one of the threads (it seems)

Can you show the stacktrace of this thread? It's possible on Visual Studio and Rider, not sure about others.

**Comment by satial-interfaces** [comment]
Yes, I can get a stack trace in Visual Studio Code. But: all threads are running, and cannot be paused when exiting the 11.0 app. It can only be terminated. I've tried Rider (2022.3 EAP 8), but the same issue is there as well. All threads are running and cannot be paused. 

Maybe someone can check this on Windows? It might be a platform problem.

**Comment by timunie** [comment]
I wonder if anything else is missing or wrong. I think it would be great if you can share your project with us. So we can check it on our own.
